### ISSUE TITLE
Elasticsearch 7.15.1 warning "request accesses aliases with names reserved for system indices"

### ISSUE THREAD

**Issue by zyqxd** [ankane/searchkick#1528]
**First**
Searchkick 4.6.1
Elasticsearch 7.15.1 (Elastic cloud)

**Describe the bug**
Warning occurs during reindexing (and maybe other tasks)
```
warning: 299 Elasticsearch-7.15.1-83c34f456ae29d60e94d886e455e6a3409bba9ed "this request accesses aliases with names reserved for system indices: [.fleet-policies-leader, .kibana_7.15.1, .kibana_task_manager, .fleet-agents, .fleet-enrollment-api-keys, .kibana_task_manager_7.15.1, .security-tokens, .fleet-servers, .security, .kibana, .fleet-policies], but in a future major version, direct access to system indices and their aliases will not be allowed"
```

**To reproduce**
rails searchkick:reindex:all

**Comment by ankane** [comment]
Hey @zyqxd, what version of the `elasticsearch` gem are you on?

**Comment by zyqxd** [comment]
here is a bit more about my `Gemfile.lock`

```
    elasticsearch (7.13.3)
      elasticsearch-api (= 7.13.3)
      elasticsearch-transport (= 7.13.3)
    elasticsearch-api (7.13.3)
      multi_json
    elasticsearch-transport (7.13.3)
```

**Comment by ankane** [comment]
Hmm, found and fixed a similar warning with searching (0d18831ae96988ed7089ffc85cd0ee36952ce254), but unable to reproduce that one. You can try enabling logging to see which call is causing it (only recommended in development to prevent logging potentially sensitive info).

```ruby
Searchkick.client.transport.logger = ActiveSupport::Logger.new(STDERR)
```

**Comment by zyqxd** [comment]
@ankane 

By similar do you mean a warning but not specifically warning 299? looks like you just commited 0d18831ae96988ed7089ffc85cd0ee36952ce254

Here is the log
```
GET https://<host>/anthologies_stories_development%2A/_alias [status:200, request:0.133s, query:n/a]
< {"anthologies_stories_development_20211109133307009":{"aliases":{"anthologies_stories_development":{}}}}
warning: 299 Elasticsearch-7.15.1-83c34f456ae29d60e94d886e455e6a3409bba9ed "this request accesses aliases with names reserved for system indices: [.fleet-policies-leader, .kibana_7.15.1, .kibana_task_manager, .fleet-agents, .fleet-enrollment-api-keys, .kibana_task_manager_7.15.1, .security-tokens, .fleet-servers, .security, .kibana, .fleet-policies], but in a future major version, direct access to system indices and their aliases will not be allowed"
```

I'm assuming the warning is coming from the `/_alias` call? doesn't look like any of the reserved names are used in the request though

**Comment by ankane** [comment]
The warning fixed in that commit deals with searching rather than indexing and had slightly different wording:

```txt
warning: 299 Elasticsearch-7.15.1-83c34f456ae29d60e94d886e455e6a3409bba9ed "this request accesses system indices: [...], but in a future major version, direct access to system indices will be prevented by default"
```

I'm not sure why the `/_alias` call would still be trying to access system indices (this was previously addressed in 54575f1b5edef97f6d97640fccff586139bf48b0), but you could try asking Elastic. I don't have more time to spend on it (and am not exactly sure how to reproduce it), but happy to revisit if someone solves it.

**Comment by zyqxd** [comment]
Seems like an issue on elastic cloud's side. Contacted their team; this is their advice

```
so these are deprecation warings since we are planning to make changes for how users interact with system indexs and their aliases. 
For now, we can hid those warnings with the following command, however you would want to enable them again before going through an upgrade to check if anything is being reported.

PUT _cluster/settings { "persistent": { "logger.org.elasticsearch.deprecation": "error" } }
```

Something to note in the searchkick upgrade plan. For now will close this issue
